File: 1470/CPP/One.cpp
#include <vector>
#include <iostream>

std::vector<int> shuffle(std::vector<int> &nums, int n)
{
    std::vector<int> ans(2 * n, 0);
    int i = 0;
    while (i < 2 * n)
    {
        ans[i] = nums[i / 2];
        ans[i + 1] = nums[n + i / 2];
        i += 2;
    }
    return ans;
}

int main()
{
    std::vector<int> nums = {2, 5, 1, 3, 4, 7};
    int n = 3;
    std::vector<int> ans = shuffle(nums, n);
    for (int i = 0; i < ans.size(); i++)
    {
        std::cout << ans[i] << " ";
    }
    return 0;
}

File: 1470/JavaScript/One.js
/**
 * @param {number[]} nums
 * @param {number} n
 * @return {number[]}
 */
var shuffle = function (nums, n) {
  let ans = [];
  for (let i = 0; i < 2 * n; i += 2) {
    ans[i] = nums[i / 2];
  }
  for (let i = 1; i < 2 * n; i += 2) {
    ans[i] = nums[n + ((i / 2) | 0)];
  }
  return ans;
};
